File: LLM-Translations/GPT-4/codenet/C++/Go/s469969316.go

package main

import (
	"fmt"
	"strconv"
	"strings"
)

func not(x int) int {
	return 2 - x
}

func and(x, y int) int {
	if x*y == 0 {
		return 0
	}
	if x == 2 && y == 2 {
		return 2
	}
	return 1
}

func or(x, y int) int {
	if x == 2 || y == 2 {
		return 2
	}
	if x == 1 || y == 1 {
		return 1
	}
	return 0
}

func parse(ss string, it *int) int {
	c := ss[*it]
	*it++

	if '0' <= c && c <= '2' {
		return int(c - '0')
	}
	if c == '-' {
		return not(parse(ss, it))
	}

	x := parse(ss, it)
	c = ss[*it]
	*it++
	y := parse(ss, it)
	*it++
	if c == '*' {
		return and(x, y)
	}
	return or(x, y)
}

func main() {
	var input, ss string
	var a [3]int
	var it, cnt int

	for {
		fmt.Scan(&input)

		if input == "." {
			break
		}

		for i := 0; i < 3; i++ {
			a[i] = i
		}
		cnt = 0

		for i := 0; i < 3; i++ {
			for j := 0; j < 3; j++ {
				for k := 0; k < 3; k++ {
					a[0] = i
					a[1] = j
					a[2] = k
					ss = input
					for index, c := range ss {
						if c == 'P' {
							ss = ss[:index] + strconv.Itoa(a[0]) + ss[index+1:]
						}
						if c == 'Q' {
							ss = ss[:index] + strconv.Itoa(a[1]) + ss[index+1:]
						}
						if c == 'R' {
							ss = ss[:index] + strconv.Itoa(a[2]) + ss[index+1:]
						}
					}
					it = 0
					if parse(ss, &it) == 2 {
						cnt++
					}
				}
			}
		}
		fmt.Println(cnt)
	}
}

// 

File: LLM-Translations/GPT-4/codenet/C++/Python/s469969316.py
import sys

def Not(x):
    return 2-x

def And(x, y):
    if x*y == 0:
        return 0
    if x == 2 and y == 2:
        return 2
    return 1

def Or(x, y):
    if x == 2 or y == 2:
        return 2
    if x == 1 or y == 1:
        return 1
    return 0

def parse(ss, it):
    c = ss[it]
    it += 1
    if '0' <= c <= '2':
        return int(c), it
    if c == '-':
        res, it = parse(ss, it)
        return Not(res), it

    x, it = parse(ss, it)
    c = ss[it]
    it += 1
    y, it = parse(ss, it)
    it += 1
    return (And(x, y) if c == '*' else Or(x, y)), it

while True:
    input_str = input().strip()
    if input_str == ".":
        break

    a = [0, 0, 0]
    cnt = 0
    for i in range(3):
        for j in range(3):
            for k in range(3):
                a[0] = i
                a[1] = j
                a[2] = k
                ss = input_str
                ss = ss.replace('P', str(a[0]))
                ss = ss.replace('Q', str(a[1]))
                ss = ss.replace('R', str(a[2]))
                it = 0
                res, _ = parse(ss, it)
                if res == 2:
                    cnt += 1
    print(cnt)


